exercises/sgf-parsing/test/Tests.hs

```diff
@@ -1,89 +1,39 @@
-    in testCase label (expected @=? parseSgf input)
+import Sgf (parseSgf)
 
 main :: IO ()
-main = exitProperly $ runTestTT $ TestList
-       [ TestList sgfParsingTests ]
-
-t :: [(Text, [Text])] -> [SgfTree] -> SgfTree
-t n = Node (Map.fromList n)
-
--- Tree, single child
-ts :: [(Text, [Text])] -> SgfTree -> SgfTree
-ts n tr = Node (Map.fromList n) [tr]
-
--- Tree, no children
-tn :: [(Text, [Text])] -> SgfTree
-tn n = Node (Map.fromList n) []
-
-sgfParsingTests :: [Test]
-sgfParsingTests =
-  [ tps "" $
-    Nothing 
-  , tps "()" $
-    Nothing
-  , tps ";" $
-    Nothing
-  , tps "(;)" $
-    Just $ tn []
-  , tps "(;A[B])" $
-    Just $ tn [("A", ["B"])]
-  , tps "(;a)" $
-    Nothing
-  , tps "(;a[b])" $
-    Nothing
-  , tps "(;Aa[b])" $
-    Nothing
-  , tps "(;A[B];B[C])" $ 
-    Just $ ts [("A", ["B"])] (tn [("B", ["C"])])
-  , tps "(;A[B](;B[C])(;C[D]))" $ 
-    Just $ t [("A", ["B"])]
-        [ tn [("B", ["C"])]
-        , tn [("C", ["D"])]
-        ]
-  , tps "(;A[b][c][d])" $
-    Just $ tn [("A", ["b", "c", "d"])]
-  , tps "(;A[\\]b\nc\\\nd\t\te\\\\ \\\n\\]])" $
-    Just $ tn [("A", ["]b cd  e\\ ]"])]
-  ]
+main = hspecWith defaultConfig {configFastFail = True} specs
+
+specs :: Spec
+specs = describe "sgf-parsing" $
+          describe "parseSgf" $ for_ cases test
+  where
+
+    test (input, expected) = it description assertion
+      where
+        description = unwords ["parse", show input]
+        assertion   = parseSgf input `shouldBe` fmap fromList <$> expected
+
+    -- As of 2016-09-18, there was no reference file
+    -- for the test cases in `exercism/x-common`.
+
+    cases = [ (""                                   , Nothing                                                                         )
+            , ("()"                                 , Nothing                                                                         )
+            , (";"                                  , Nothing                                                                         )
+            , ("(;)"                                , Just  $ Node [] []                                                              )
+            , ("(;A[B])"                            , Just  $ Node [("A", ["B"])] []                                                  )
+            , ("(;a)"                               , Nothing                                                                         )
+            , ("(;a[b])"                            , Nothing                                                                         )
+            , ("(;Aa[b])"                           , Nothing                                                                         )
+            , ("(;A[B];B[C])"                       , Just  $ Node [("A", ["B"])] [ Node [("B", ["C"])] []                          ] )
+            , ("(;A[B](;B[C])(;C[D]))"              , Just  $ Node [("A", ["B"])] [ Node [("B", ["C"])] [] , Node [("C", ["D"])] [] ] )
```

Comment:
Changed!

I agree with you about breaking the children in multiple lines, but I did that just when there where multiple children, to keep the siblings vertically aligned (line 36):

```haskell
... , Just  $ Node [("A", ["B"])] [ Node [("B", ["C"])] [] ] )
... , Just  $ Node [("A", ["B"])] [ Node [("B", ["C"])] []
                                  , Node [("C", ["D"])] [] ] )
```

Is that OK for your?

*ps: It's hard to format trees...*